libs/common/types.hpp

```diff
@@ -89,6 +90,17 @@ namespace iroha {
       }
       return res;
     }
+
+    static blob_t<size_> from_string(const std::string &data) {
+      if(data.size() != size_) {
+        throw BadFormatException("blob_t: input string has incorrect length");
```

Comment:
Please replace this with `nonstd::nullopt`.

Reply:
it will be replaced later with `result`/`expected`

Reply:
Great, so we can later change the return type from `nonstd::optional<blob_t<size_>>` to `result<...>` or `expected<...>` and the equivalent of `nonstd::nullopt`.
Exception has to be removed anyway.